Moonshot/player/player.gd

```diff
@@ -168,7 +165,10 @@ func take_damage(damage, attack_dir, is_damage_tile: bool = false) -> void:
 	if not invulnerable:
 		Utils.player_stats.health -= damage
 		if Utils.IS_DEBUG: print("Attack Dir : ", attack_dir)
-		
+		if Utils.player_stats.health <= 0:
+			on_death()
+			safety = true
+			return
```

Comment:
Precede the stagger and invuln logic with death. This disables shooting now when we die.